src/utils/validations/component/validations/with-validation.hoc.js

```diff
@@ -0,0 +1,159 @@
+      if (result && types.includes('warnings')) result = await this.runValidation('warnings');
+      if (result && types.includes('info')) result = await this.runValidation('info');
+      return result;
+    }
+
+    runValidation = (type) => {
+      return new Promise((resolve) => {
+        if (this.props[type].length === 0) {
+          resolve(true);
+        }
+        const validate = validator(this.props[type]);
+        validate(this.props.value)
+          .then(() => {
+            switch (type) {
+              case 'validations':
+                if (this.state.hasError !== false && this.context.adjustCount) {
+                  this.setState({ hasError: false });
+                  this.context.adjustCount('error', -1);
+                }
+                break;
+              case 'warnings':
+                if (this.state.hasWarning !== false && this.context.adjustCount) {
+                  this.setState({ hasWarning: false });
+                  this.context.adjustCount('warning', -1);
+                }
+                break;
+              case 'info':
+                if (this.state.hasInfo !== false && this.context.adjustCount) {
+                  this.setState({ hasInfo: false });
+                  this.context.adjustCount('info', -1);
+                }
+                break;
+              default:
+            }
+            resolve(true);
+          })
+          .catch((e) => {
+            switch (type) {
+              case 'validations':
+                this.setState({ hasError: e });
+                if (this.context.adjustCount) this.context.adjustCount('error', 1);
+                break;
+              case 'warnings':
+                this.setState({ hasWarning: e });
+                if (this.context.adjustCount) this.context.adjustCount('warning', 1);
+                break;
+              case 'info':
+                this.setState({ hasInfo: e });
+                if (this.context.adjustCount) this.context.adjustCount('info', 1);
+                break;
+              default:
+            }
+            resolve(false);
+          });
+      });
+    }
+
+    renderValidationMarkup = () => {
+      let type;
+      if (this.state.hasInfo) type = 'info';
+      if (this.state.hasWarning) type = 'warning';
+      if (this.state.hasError) type = 'error';
+      if (type) {
+        if (Array.isArray(this.props.children)) {
+          return [...this.props.children, <Icon key={ `${type}-icon` } type={ type } />];
+        }
+        return [this.props.children, <Icon key={ `${type}-icon` } type={ type } />];
+      }
+      return this.props.children;
+    }
+
+    handleBlur = (ev) => {
+      this.validate();
+      if (this.props.onBlur) this.props.onBlur(ev);
+    }
+
+    render() {
+      return (
+        <WrappedComponent
+          onBlur={ this.handleBlur }
+          { ...this.props }
+        >
+          { this.renderValidationMarkup() }
```

Comment:
I think that we could extract the children prop from the `renderValidationMarkup` method:
```
{ this.props.children }
{ this.renderValidationMarkup() }
```

Reply:
Agreed, I've made the changes as it does seem unnecessary to have it handled in `renderValidationMarkup`